source/extensions/omni.isaac.lab/omni/isaac/lab/scene/interactive_scene.py

```diff
@@ -90,20 +90,28 @@ def __init__(self, cfg: InteractiveSceneCfg):
         self.cloner = GridCloner(spacing=self.cfg.env_spacing)
         self.cloner.define_base_env(self.env_ns)
         self.env_prim_paths = self.cloner.generate_paths(f"{self.env_ns}/env", self.cfg.num_envs)
+        # environment origins
+        self._default_env_origins = None
         # create source prim
         self.stage.DefinePrim(self.env_prim_paths[0], "Xform")
-        # clone the env xform
-        env_origins = self.cloner.clone(
-            source_prim_path=self.env_prim_paths[0],
-            prim_paths=self.env_prim_paths,
-            replicate_physics=False,
-            copy_from_source=True,
-        )
-        self._default_env_origins = torch.tensor(env_origins, device=self.device, dtype=torch.float32)
+        # when replicate_physics=False, we assume heterogeneous environments and clone the xforms first.
+        # this triggers per-object level cloning in the spawner.
+        if not self.cfg.replicate_physics:
+            # clone the env xform
+            env_origins = self.cloner.clone(
+                source_prim_path=self.env_prim_paths[0],
+                prim_paths=self.env_prim_paths,
+                replicate_physics=False,
+                copy_from_source=True,
+            )
+            self._default_env_origins = torch.tensor(env_origins, device=self.device, dtype=torch.float32)
         self._global_prim_paths = list()
```

Comment:
When user calls `filter_collisions(global_prim_paths)` should we store the global prim paths from that call into this list?